### PR TITLE
Add goroute framework

### LINKED ISSUE #1684: Add route
Hi @anjmao,

Would you mind to add (or consent that we add) [route](https://github.com/goroute/route) in `here` ?

This project compare performance based on 3 routes :

+ `GET` `/`, **SHOULD** return a successful status code and an empty string
+ `GET` `/user/:id`, **SHOULD** return a successful status code and the id
+ `POST` `/user`, **SHOULD** return a successful status code and an empty string

Regards,

### PR DESCRIPTION
**Pull request by anjmao**
Before submitting your PR, please review the following checklist :

## If you are adding a framework

+ [x] A `Dockerfile` exists, or can be built ?
+ [x] I've updated `CI` config file (and **commited** `.travis.yml`)
~~~sh
bin/make ci_config
~~~
+ [x] I've updated `.dependabot` config file  (and **commited** `.dependabot/config.yml`)
~~~sh
bin/make deps_config
~~~
+ [x] All tests passes ?
~~~
export FRAMEWORK=<MY_FRAMEWORK>
shards install
shards build --release
bin/make config
bin/neph ${FRAMEWORK}
crystal spec
~~~

Closes #1684

### CODE REVIEW

**Comment by anjmao** [comment]
Closes https://github.com/the-benchmarker/web-frameworks/issues/1684

File: go/goroute/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+FROM golang:1.12.8 AS buildenv
+
+ENV GO111MODULE=on
```
Comment (anjmao) [inline go/goroute/Dockerfile]: @waghanza I also added GO111MODULE=on which is missing in almost all go web frameworks. If this variable is not set and working dir is set to /go/src/app go will treat this as old GOPATH build setup and will not use go modules correctly.

File: go/goroute/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ENV CGO_ENABLED=0
+ENV GOOS=linux
+ENV GOARCH=amd64
+ENV GOPROXY=https://proxy.golang.org
```
[the remark this answers is not in the document]
Comment (anjmao) [reply go/goroute/Dockerfile]: By default go will fetch modules from source control provider (github, gitlab) etc. When setting  this variable it calls proxy cache server which is faster. https://proxy.golang.org is hosted by Google. I can update other Dockerfiles too.

File: go/goroute/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+WORKDIR /go/src/app
```
[the remark this answers is not in the document]
Comment (anjmao) [reply go/goroute/Dockerfile]: WORKIR should not point to binary. For go apps `/go/src/...` is kind a default path. It also used in all other examples.

File: go/goroute/Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (anjmao) [reply go/goroute/Dockerfile]: By default go will fetch modules from source control provider (github, gitlab) etc. When setting this variable it calls proxy cache server which is faster. https://proxy.golang.org is hosted by Google. I can update other Dockerfiles too.

File: go/goroute/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+ENV GOOS=linux
+ENV GOARCH=amd64
+
+WORKDIR /app
+ADD main.go go.mod go.sum ./
+
+RUN GOPROXY=https://proxy.golang.org go mod download
+RUN go build -a -ldflags '-extldflags "-static"' -o /app/do ./
+
+FROM scratch
```
[the remark this answers is not in the document]
Comment (anjmao) [reply go/goroute/Dockerfile]: For go apps when building static binary it is preferable to simply use scratch as a base image. It's also used in other examples.

**Comment by anjmao** [comment]
@waghanza I suggest such Dockerfile for all go apps. I can update other examples too.

```Dockerfile
FROM golang:1.12.8 AS buildenv

ENV GO111MODULE=on
ENV CGO_ENABLED=0
ENV GOOS=linux
ENV GOARCH=amd64
ENV GOPROXY=https://proxy.golang.org

WORKDIR /go/src/app
ADD main.go go.mod ./

RUN go get
RUN go build -a -ldflags '-extldflags "-static"' -o /go/bin/app ./

FROM scratch
COPY --from=buildenv /go/bin/app /go/bin/app
CMD ["/go/bin/app"]
``

File: go/goroute/Dockerfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (waghanza) [reply go/goroute/Dockerfile]: Sure this a kind of default path

However, I'll extract this binary for a local execution (next feature), but I can accept like this

But I'll update this dockerfile

**Comment by waghanza** [comment]
Why not 😁

I'm not `go` specialist

What does `go mod download` is for ?

I prefer using **alpine** instead of **scratch** for docker, since with **alpine** binary file could be downloaded from container (this could not be done  in **scratch**)

**Pull request merged**